src/brushes/pencil_brush.class.js

```diff
@@ -21,6 +35,41 @@
     initialize: function(canvas) {
       this.canvas = canvas;
       this._points = [];
+      this._attachKeyboardListeners();
+    },
+
+    /**
+     * Listens to `shift` key press
+     * @private
+     * @returns disposer
+     */
+    _attachKeyboardListeners: function () {
+      var _this = this;
+      var setShift = function (opt) {
+        if (opt.e.shiftKey) {
+          opt.e.preventDefault();
+          _this.drawStraightLine = true;
+        }
+        else {
+          _this.drawStraightLine = false;
+        }
+      };
+      var revokeShift = function () {
+        _this.drawStraightLine = false;
+      };
+      this.canvas.on('mouse:down', setShift);
+      this.canvas.on('mouse:move', setShift);
+      this.canvas.on('mouse:up', revokeShift);
+      this._detach = function disposer() {
+        this.canvas.off('mouse:down', setShift);
+        this.canvas.off('mouse:move', setShift);
+        this.canvas.off('mouse:up', revokeShift);
+      };
+      return this._detach;
+    },
```

Comment:
ok sorry i took my time to review this, but there was a whole thing i wanted to try.
In my opinion all this new function is not needed. You do not need to attach keyboard events at all to grab the shift key.

Reply:
https://github.com/fabricjs/fabric.js/pull/7034/files#diff-62b861bab6725568506ad50661f0be7c7d3d79b8ba69229f664226fb54d9269bR93

Can we add here for example:
```
this.drawStraightLine = options.e[this.straightLineKey];
```

and same on mouseMove / mouseUp.

Keeping `this.straightLineKey` default to 'shiftKey' will allow people to disable the functionality setting it to empty space or a key that does not exist.

Reply:
54f4272
Works!